overlord/snapstate/snapstate.go

```diff
@@ -588,6 +593,14 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	addTask(autoConnect)
 	prev = autoConnect
 
+	if deviceCtx.HasModeenv() && snapsup.Type == snap.TypeKernel {
+		// This task needs to run after the restart triggered by
+		// link-snap for kernel snaps.
```

Comment:
I'd slightly reword the comment:
```suggestion
		// This task needs to run after we're back and running the new kernel 
		// after a reboot was requested in link-snap handler for kernel snaps
```